File: 数据结构专题/88_合并两个有序数组.py


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        nums1[m:] = nums2
        #nums1 = sorted(nums1) #sorted返回的是一个新的列表，题目要求nums1
        nums1.sort() #直接改变nums1的元素顺序

File: 数据结构专题/88_合并两个有序数组.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
	int i = 0;
	int _callback(const void* num1, const void* num2)
	{
		return (*(int*)num1 - *(int*)num2);
	}
	for (i = 0; i < n; i++)
	{
		nums1[i + m] = nums2[i];
	}
	qsort(nums1, nums1Size, sizeof(nums1[0]), _callback);
}